source/blog/2015-08-16-ember-2-1-beta-released.md

```diff
@@ -0,0 +1,171 @@
+  }
+});
+```
+
+In this example, all warnings are silenced:
+
+```js
+// next is not called, so no warnings get the default behavior
+Ember.Debug.registerWarnHandler(() => {});
+```
+
+Handlers are provided with the following data through arguments:
+
+* `message` is the text logged by default
+* `next` is a caller for the previously registered handler. Often, this is the
+  default handler.
+* `options.id` is an id in the format `package-name.specific-deprecation`
+
+Deprecation handlers will also be provided:
+
+* `options.until` is the version of Ember this feature and deprecation will be
+  removed in
+
+As of Ember 1.13, `deprecate` and `warn` calls must provide an `id` option,
+and `deprecate` calls must additionally provide an `until` option.
+Addons not providing this data during
+2.x will trigger a deprecation warning.
+
+For more information see [RFC #65](https://github.com/emberjs/rfcs/blob/master/text/0065-deprecation-warning-handlers.md)
+and the implementation in [PR #11833](https://github.com/emberjs/ember.js/pull/11833).
+This API can be used with older versions of Ember via the
+[ember-debug-handlers-polyfill](https://github.com/rwjblue/ember-debug-handlers-polyfill),
+though `id` and `until` data is not provided until Ember 1.13.
+
+#### Registry and Container Reform
+
+The Ember.js registry and container are some of the most extensively used
+private APIs in the framework. They
+provided one of the only ways to lookup arbitrary objects from Ember's
+dependency container.
+
+In 2.x, we are committed to stabilizing this part of the framework and
+offering public APIs. This first step creates a normalized way to interact
+with `register` and `lookup` that we expect to last through the 2.x cycle
+and beyond.
+
+`Ember.Application` instances are passed as the first argument to `initializer`
+hooks in 2.1. `initializer` hooks are where dependencies between object types
+can be configured, and factories can be registered. Several public APIs will
+exist on `Ember.Application` instances, some of them new:
+
+* `register` - register a factory
+* `inject` - inject a factory into another factory, or all factories of a type
+* `unregister` - remove a factory from registration
+* `resolveRegistration` - fetch a registed factory
+* `hasRegistration` - check for a registered factory
+* `registerOption`, `registeredOption`, `registerOptions`, `registeredOptions`,
+  `registerOptionsForType`, `registeredOptionsForType` which manage options
+  for a factory (is it a singleton, can it be instantiated).
+
+`Ember.ApplicationInstance` instances are passed as the first argument to
+`instanceInitializer` hooks in 2.1. `instanceInitializer` hooks are where
+objects and classes can be fetched out of the configured and booted application.
+Two new relevent public APIs will exist on `Ember.ApplicationInstance`s:
+
+* `lookup` - fetch an instance of a factory (with dependencies)
+* `lookupFactory` - fetch a factory (with dependencies)
```

Comment:
Ah private in master, gotcha: https://github.com/emberjs/ember.js/blob/master/packages/ember-runtime/lib/mixins/container_proxy.js#L67